internal/services/containers/container_registry_task_resource.go

```diff
@@ -0,0 +1,1573 @@
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.RegistryID,
+		},
+		"platform_setting": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*schema.Schema{
+					"os": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(legacyacr.Windows),
+							string(legacyacr.Linux),
+						}, false),
+					},
+					"architecture": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(legacyacr.Amd64),
+							string(legacyacr.Arm),
+							string(legacyacr.Arm64),
+							string(legacyacr.ThreeEightSix),
+							string(legacyacr.X86),
+						}, false),
+					},
+					"variant": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(legacyacr.V6),
+							string(legacyacr.V7),
+							string(legacyacr.V8),
+						}, false),
+					},
+				},
+			},
+		},
+		"docker_step": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*schema.Schema{
+					"dockerfile_path": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"context_path": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"context_access_token": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						Sensitive:    true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"image_names": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeString,
+						},
+					},
+					"is_push_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						Default:  true,
+					},
+					"is_cache_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						Default:  true,
+					},
+					"target": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"arguments": {
+						Type:     pluginsdk.TypeMap,
+						Optional: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeString,
+						},
+					},
+					"secret_arguments": {
+						Type:      pluginsdk.TypeMap,
+						Optional:  true,
+						Sensitive: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeString,
+						},
+					},
+				},
+			},
+			ConflictsWith: []string{"file_task_step", "encoded_task_step"},
+		},
+		"file_task_step": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*schema.Schema{
+					"task_file_path": {
```

Comment:
task and file are already in the resource and block names?
```suggestion
					"path": {
```

Reply:
Shall we be more explict that this *path* here is the task file's path, rather than some other path (e.g. value file path, context path, etc.)?

Reply:
I still feel `task_file_path` is better with the presence of `value_file_path` in the same block.